WebDriverAgentLib/Routing/FBResponseJSONPayload.m

```diff
@@ -20,17 +20,77 @@ @interface FBResponseJSONPayload ()
 
 @implementation FBResponseJSONPayload
 
+
+/*
+ * Convenience method to do the check and validation in one.
+ */
++ (NSString*) makeValidUTF8:(NSString*) stringToCheck
+{
+  if (![FBResponseJSONPayload isValidUTF8 :stringToCheck])
+  {
+    return [FBResponseJSONPayload removeInvalidCharsFromString:stringToCheck];
+  }
+  else
+  {
+    return stringToCheck;
+  }
+}
+
+/*
+ * Returns true if the string can be converted to UTF8
+ */
++ (int) isValidUTF8:(NSString*) stringToCheck
+{
+  return ([stringToCheck UTF8String] != nil);
+}
+
+/*
+ * Removes invalid UTF8 chars from the NSString
+ * This method is slow, so only run it on strings that fail the isValidUTF8 check.
+ */
++ (NSString*) removeInvalidCharsFromString:(NSString*) stringToCheck
+{
+  NSMutableString* fixedUp = [NSMutableString stringWithString:@""];
```

Comment:
Perhaps something like [that](https://stackoverflow.com/questions/10932405/nsstring-easy-way-to-remove-utf-8-accents-from-a-string) would be faster

Reply:
In my case, I was viewing a Korean webpage, so it probably did not contain accents or umlauts. However I will take a look if there is any other builtin methods to make it more simple and fast. Thank you.